src/main/java/com/faforever/api/user/GogService.java

```diff
@@ -0,0 +1,177 @@
+package com.faforever.api.user;
+
+import com.faforever.api.config.FafApiProperties;
+import com.faforever.api.data.domain.User;
+import com.faforever.api.error.ApiException;
+import com.faforever.api.error.ErrorCode;
+import com.google.gson.Gson;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class GogService {
+  // TODO: what characters to allow?
+  private static final Pattern GOG_USERNAME_PATTERN = Pattern.compile("[a-z\\d-_!?.]+"); // gets lower cased by controller, gog doesn't care about capitalization, this causes db entries to be unique
+  private static final Pattern PROFILE_USER_STATUS_PATTERN = Pattern.compile("window\\.profilesData\\.profileUserPreferences\\s=\\s\\{\"bio\":\"(.*?)\"");
```

Comment:
Use the multiline string (`"""`) and avoid some of the escaping

Reply:
I take this back. The Java implementation version doesn't work here at all (single line and ending with quotes just breaks)